### PR TITLE
typeclasses in spray-json

### PR DESCRIPTION
**Pull request by hibikir**
What's left: We do not have good support for multiple links for multiple authors: If I uncomment Jess' details, it just displays whoever's are last. @chrisshafer should probably be able come up with something to make collaborations work.

### CODE REVIEW

**Comment by ddgenome** [comment]
@slbahr look good to you?

File: _posts/2015-09-23-implicits-typeclasses.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+<pre>
+    {{"implicit def"|sc: "implicitdef"}} listFormat[{{"T : JsonFormat"| sc: "contextbound"}}] = new RootJsonFormat[List[T]] {
+      def write(list: List[T]) = ..
+      def read(value: {{"JsValue"| sc: "jsValue"}}): List[T] = ..
+    }
+</pre>
+
+What is this doing? First, we have to understand some new syntax: inside the {{"type parameter, there is a colon, followed by a type class"| sc: "contextbound"}}. 
+This is called [Context Bounds](http://docs.scala-lang.org/tutorials/FAQ/context-and-view-bounds.html) (good luck finding the documentation without knowing this special name).
+This is shorthand for "{{"a type T such that there exists in the magic hat a JsonFormat[T]"| sc: "contextbound"}}]".
```
Comment (ddgenome) [inline _posts/2015-09-23-implicits-typeclasses.md]: There is an extra closing square bracket. Near the end of the line.

File: _posts/2015-09-23-implicits-typeclasses.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+The context-bounds notation above expands to:
+
+<pre>
+    implicit def listFormat[{{"T"| sc: "contextbound"}}]({{"implicit _ : JsonFormat[T]"| sc: "contextbound"}}) = new RootJsonFormat[List[T]] {
+      def write(list: List[T]) = ..
+      def read(value: {{"JsValue"| sc: "jsValue"}}): List[T] = ..
+    }
+</pre>
+
+The implicit parameter that the write function inside listFormat will be able to call {{".toJson"| sc: "toJson""}} on the elements in the List.
```
Comment (ddgenome) [inline _posts/2015-09-23-implicits-typeclasses.md]: "The implicit parameter [ensures] that the write function..."

File: _posts/2015-09-23-implicits-typeclasses.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+This {{"implicit def"|sc: "implicitdef"}} does not work the same way as a [view](http://engineering.monsanto.com/2015/07/31/implicit-conversions/), which converts a single type to another.
+Instead, it is is a supplier of implicit values. It can give the compiler a JsonFormat[List[T]],as long as the compiler supplies a JsonFormat[T].
```
Comment (ddgenome) [inline _posts/2015-09-23-implicits-typeclasses.md]: "It is is a supplier" delete the second "is".

**Comment by slbahr** [comment]
This is approved.

**Pull request merged**